File: simpleWebSocket/src/main/java/com/sushant/websocket/service/ChatroomService.java
/**
 * Filename: ChatroomService.java
 * -----------------------------------------
 * Licensed: MIT License https://opensource.org/licenses/MIT
 */
package com.sushant.websocket.service;

import com.sushant.websocket.model.ChatRooms;

/**
 * @author Sushant
 *
 */
public interface ChatroomService {

    /**
     * Get All ChatRooms
     * @return {@link ChatRooms}
     */
    ChatRooms getAllChatRooms();
}

File: simpleWebSocket/src/main/webapp/static/scripts/services/chatRoomService.js
/**
 * filename : chatRoomService.js
 * -------------------------------
 * Licensed Under MIT.
 * 
 * ChatRoom Service for managing chat rooms
 */
app.service('ChatRoomService', function($http, $q) {


	/**
	 * Fetch all available chat rooms
	 */
	
	this.getAllChatRooms = function() {
		return $http.get("chatrooms/")
		.then(function(result){							
			return result.data;
		});
	}
		
});